tools/methyldackel/MethylDackel.xml

```diff
@@ -104,301 +110,487 @@
+    
+            <data  name="outFileExtractCHHMethylKit" format="txt" from_work_dir="output_CHH.methylKit" 
+                label="${tool.name} on ${on_string}: methylKit CHH">
+                <filter>main_task['task'] == "extract"</filter>
+                <filter>main_task['extract_outputs'] and "--methylKit" in main_task['extract_outputs']</filter>
+                <filter>advanced_options['methylation_metrics'] and "--CHH" in advanced_options['methylation_metrics']</filter>
             </data>
             <data  name="outFileExtractCHHCounts" format="bedgraph" from_work_dir="output_CHH.counts.bedGraph" 
-                label="${tool.name} on ${on_string} (CHH counts)">
-                <filter>main_task['task']  == "extract" and advanced_options['counts']</filter>
-                <filter>advanced_options['options'] == "yes"</filter>
-                <filter>advanced_options['CHH']</filter>
+                label="${tool.name} on ${on_string}: counts CHH">
+                <filter>main_task['task'] == "extract"</filter>
+                <filter>main_task['extract_outputs'] and "--counts" in main_task['extract_outputs']</filter>
+                <filter>advanced_options['methylation_metrics'] and "--CHH" in advanced_options['methylation_metrics']</filter>
             </data>
             <data  name="outFileExtractCHHFraction" format="bedgraph" from_work_dir="output_CHH.meth.bedGraph" 
-                label="${tool.name} on ${on_string} (CHH fraction)">
-                <filter>main_task['task']  == "extract" and advanced_options['fraction']</filter>
-                <filter>advanced_options['options'] == "yes"</filter>
-                <filter>advanced_options['CHH']</filter>
+                label="${tool.name} on ${on_string}: fraction CHH">
+                <filter>main_task['task'] == "extract"</filter>
+                <filter>main_task['extract_outputs'] and "--fraction" in main_task['extract_outputs']</filter>
+                <filter>advanced_options['methylation_metrics'] and "--CHH" in advanced_options['methylation_metrics']</filter>         
             </data>
             <data  name="outFileMbiasTSV" format="tabular" from_work_dir="out_mbias.tsv"
-                label="${tool.name} on ${on_string} (methylation bias, tsv)">
+                label="${tool.name} on ${on_string}: methylation bias (tsv)">
                 <filter>main_task['task']  == 'mbias'</filter>
                 <filter>main_task['tsv']</filter>
             </data>
+
             <data  name="outFileMbiasCpGOT" format="svg" from_work_dir="out_mbias_OT.svg"
-                label="${tool.name} on ${on_string} (methylation bias, original top strand)">
+                label="${tool.name} on ${on_string}: methylation bias (original top strand)">
                 <filter>main_task['task']  == 'mbias'</filter>
             </data>
             <data  name="outFileMbiasCpGOB" format="svg" from_work_dir="out_mbias_OB.svg"
```

Comment:
I'm not sure if those files are correctly filtered, but I didn't modify them because not sure if all those should be generated by default or not.